haystack/nodes/prompt/prompt_node.py

```diff
@@ -490,40 +526,37 @@ def invoke(self, *args, **kwargs):
             "best_of": kwargs_with_defaults.get("best_of", 1),
             "logit_bias": kwargs_with_defaults.get("logit_bias", {}),
         }
-        headers = {"Authorization": f"Bearer {self.api_key}", "Content-Type": "application/json"}
-        response = requests.request(
-            "POST",
-            self.url,
-            headers=headers,
-            data=json.dumps(payload),
-            timeout=float(os.environ.get(HAYSTACK_REMOTE_API_TIMEOUT_SEC, 30)),
-        )
-        res = json.loads(response.text)
+        res = openai_request(url=self.url, api_key=self.api_key, payload=payload)
+        _check_openai_text_completion_answers(result=res, payload=payload)
+        responses = [ans["text"].strip() for ans in res["choices"]]
+        return responses
 
-        if response.status_code != 200:
-            openai_error: OpenAIError
-            if response.status_code == 429:
-                openai_error = OpenAIRateLimitError(f"API rate limit exceeded: {response.text}")
-            else:
-                openai_error = OpenAIError(
-                    f"OpenAI returned an error.\n"
-                    f"Status code: {response.status_code}\n"
-                    f"Response body: {response.text}",
-                    status_code=response.status_code,
-                )
-            raise openai_error
+    def _ensure_token_limit(self, prompt: str) -> str:
+        """Ensure that length of the text is within the maximum length of the model."""
+        n_prompt_tokens = count_openai_tokens(prompt, self._tokenizer, use_tiktoken=USE_TIKTOKEN)
```

Comment:
This function will truncate the end of the prompt text to ensure the prompt text + answer length fits within the token limit. This means that if instructions are provided at the end of a PromptTemplate they will be removed, which could cause undesired behavior from the PromptNode. For example, if we were to use this prompt
```python
prompt_text = """
Summarize the following passages.

Passages:
$example_passages

Summary:
$example_summary

Passages:
$documents

Summary:
"""
```
and the `$documents` is too long, then the `Summary:` instruction at the end of the prompt will be removed. @TuanaCelik faced this exact problem when using this branch with her demo. This is not ideal, but I'm not sure how else to do the truncation due to the flexibility in the prompts of the `PromptNode`.